### PR TITLE
Fix doc about watcher timeout

### PR DESCRIPTION
**Pull request by ramonberrutti**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Better describe the watch functionality.
After some tests, I figured out that the timeout option is bad explained.

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go [hunk 1]
```diff
@@ -367,6 +367,7 @@ type ListOptions struct {
 	ResourceVersion string `json:"resourceVersion,omitempty" protobuf:"bytes,4,opt,name=resourceVersion"`
 	// Timeout for the list/watch call.
 	// This limits the duration of the call, regardless of any activity or inactivity.
+	// This timeout don't change the client timeout.
```
Comment (caesarxuchao) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: How about just modify line 369 to "This limits the duration of the call on the server-side"

File: staging/src/k8s.io/client-go/tools/cache/reflector.go [hunk 2]
```diff
@@ -287,8 +287,8 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 		timeoutSeconds := int64(minWatchTimeout.Seconds() * (rand.Float64() + 1.0))
 		options = metav1.ListOptions{
 			ResourceVersion: resourceVersion,
-			// We want to avoid situations of hanging watchers. Stop any wachers that do not
-			// receive any events within the timeout window.
+			// We want to avoid situations of hanging watchers.
+			// Stop the watcher after the time window.
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/tools/cache/reflector.go]: How about "Stop the watchers after the timeout seconds".

**Review by caesarxuchao — COMMENTED** [review]
You need to run hack/update-all.sh to update generated API docs, otherwise the CI will fail.

**Comment by ramonberrutti** [comment]
@caesarxuchao I'm getting this error running make update.
`
can't load package: package k8s.io/kubernetes/vendor/k8s.io/code-generator/cmd/deepcopy-gen: no matching versions for query "latest"
`

**Comment by caesarxuchao** [comment]
Why are there many commits about updating dependencies? Maybe you need a rebase?

**Review by alejandrox1 — CHANGES_REQUESTED** [review]
@ramonberrutti 
thank you very much for working on this.
This PR got labeled with a ton of stuff since you modified some key files so we need some cleanup (this will also fix the test failures).
Could you please run 
* `./hack/update-generated-protobuf.sh`
* `./update-generated-swagger-docs.sh`

These can be found here https://github.com/kubernetes/kubernetes/tree/master/hack

/priority important-soon
/milestone v1.17

**Comment by ramonberrutti** [comment]
@alejandrox1 thanks, I rebase with master and run make update. Thanks again :)

**Comment by kikisdeliveryservice** [comment]
Bug Triage Release Team checking in. Code freeze for the 1.17 release is on November 18. Is this still intended for 1.17? Thanks!

**Comment by kikisdeliveryservice** [comment]
@lavalamp @ramonberrutti 

Is this still intended for 1.17 or being bumped to 1.18? Code freeze is Nov 14th.

**Comment by ramonberrutti** [comment]
@kikisdeliveryservice This is just a document fix.

**Comment by ramonberrutti** [comment]
I will rebase to fix the conflict.

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go [hunk 3]
```diff
@@ -366,7 +366,8 @@ type ListOptions struct {
 	// +optional
 	ResourceVersion string `json:"resourceVersion,omitempty" protobuf:"bytes,4,opt,name=resourceVersion"`
 	// Timeout for the list/watch call.
-	// This limits the duration of the call, regardless of any activity or inactivity.
+	// This limits the duration of the call on the server-side, regardless of any activity or inactivity.
+	// This timeout don't change the client timeout.
```
Comment (lavalamp) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: > This timeout don't change the client timeout

What is the client timeout, and how does one change it?

grammar: s/don't/doesn't/

**Comment by josiahbjorgaard** [comment]
/milestone v1.18

(Code freeze v1.17)

**Comment by jtslear** [comment]
Hello @ramonberrutti, @lavalamp, and @caesarxuchao 
Bug Triage team here for the 1.18 release. This is a friendly reminder that code freeze is scheduled for 5 March. Is this PR still intended for milestone 1.18?  I see we have a few conflicts that need to be addressed.

**Comment by jtslear** [comment]
Hello @ramonberrutti,
Bug Triage team here for the 1.18 release. This is a friendly reminder that code freeze is scheduled for 5 March. I see we have a few conflicts and requested changes that need to be addressed.  Are we still trying to accomplish this work in time for code freeze?

**Comment by ramonberrutti** [comment]
Ty for the remainder :)

**Comment by jtslear** [comment]
Due to reaching the code freeze deadline, I'm going to bump this to the next release.

/milestone v1.19

**Comment by bai** [comment]
👋 Hello from Bug Triage team! This PR has not been updated for a while, so I'd like to check its status. The code freeze is starting June 25th (about 4 weeks from now) and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.

As the PR is tagged for 1.19, is it still planned for this release?

**Pull request closed without merging**